grr/server/grr_response_server/output_plugins/elasticsearch_plugin.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+# Lint as: python3
+"""OutputPlugin that sends Flow results to an Elasticsearch cluster.
+
+Configuration values for this plugin can be found in
+core/grr_response_core/config/output_plugins.py
+
+The specification for the indexing of documents is
+https://www.elastic.co/guide/en/elasticsearch/reference/7.1/docs-index_.html
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import unicode_literals
+
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Text
+from urllib import parse as urlparse
+
+import requests
+
+from google.protobuf import json_format
+from grr_response_core import config
+from grr_response_core.lib.rdfvalues import flows as rdf_flows
+from grr_response_core.lib.rdfvalues import protodict as rdf_protodict
+from grr_response_core.lib.rdfvalues import structs as rdf_structs
+from grr_response_core.lib.util.compat import json
+from grr_response_proto import output_plugin_pb2
+from grr_response_server import data_store
+from grr_response_server import export
+from grr_response_server import output_plugin
+from grr_response_server.gui.api_plugins import flow as api_flow
+
+BULK_OPERATIONS_PATH = "_bulk"
+
+JsonDict = Dict[Text, Any]
+
+
+class ElasticsearchConfigurationError(Exception):
+  """Error indicating a wrong or missing Elasticsearch configuration."""
+  pass
+
+
+class ElasticsearchOutputPluginArgs(rdf_structs.RDFProtoStruct):
+  """An RDF wrapper class for the arguments of ElasticsearchOutputPlugin."""
+  protobuf = output_plugin_pb2.ElasticsearchOutputPluginArgs
+  rdf_deps = []
+
+
+def _ToDict(rdfval: rdf_structs.RDFProtoStruct) -> JsonDict:
+  return json_format.MessageToDict(rdfval.AsPrimitiveProto(), float_precision=8)
+
+
+class ElasticsearchOutputPlugin(output_plugin.OutputPlugin):
+  """OutputPlugin that sends Flow results to Elasticsearch."""
+
+  name = "elasticsearch"
+  description = "Send flow results to Elasticsearch."
+  args_type = ElasticsearchOutputPluginArgs
+
+  def __init__(self, *args, **kwargs):
+    """See base class."""
+    super().__init__(*args, **kwargs)
+
+    url = config.CONFIG["Elasticsearch.url"]
+    self._verify_https = config.CONFIG["Elasticsearch.verify_https"]
+    self._token = config.CONFIG["Elasticsearch.token"]
+    # Allow the Flow creator to override the index, fall back to configuration.
+    self._index = self.args.index or config.CONFIG["Elasticsearch.index"]
+
+    if not url:
+      raise ElasticsearchConfigurationError(
+          "Cannot start ElasticsearchOutputPlugin, because Elasticsearch.url"
+          "is not configured. Set it to the base URL of your Elasticsearch"
+          "installation, e.g. 'https://myelasticsearch.example.com:9200'.")
```

Comment:
I would place right below the `url = config.CONFIG["Elasticsearch.url"]` part—this way you avoid executing code that will never be used because the exception.